### ISSUE TITLE
aarch64 0.27 release - jit_jitt_array_compress_3_FAILED : Segmentation error vmState=0x00000000

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#13230]
Failure link
------------

From an internal build `Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/27/` `fuji`:
```
semeru version "1.8.0_302"
Semeru Runtime Environment (build 1.8.0_302-b08)
Eclipse OpenJ9 VM (build openj9-0.27.0, JRE 1.8.0 Linux aarch64-64-Bit Compressed References 20210721_166 (JIT enabled, AOT enabled)
OpenJ9   - 1851b0074
OMR      - 9db1c870d
JCL      - 96acb976bc based on jdk8u302-b08)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=upstream&amp;TARGET=jit_jitt_array_compress_3&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk8_j9_sanity.functional_aarch64_linux&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=AdoptOpenJDK%3Amaster&amp;TEST_PARALLELIZATION_PARAMS=&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=aarch64_linux&amp;GENERATE_JOBS=true&amp;KEEP_REPORTDIR=false&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;LABEL=&amp;TEST_OPTIONS_PARAMS=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-linux-aarch64-openj9%2F166%2FSemeru-testimage_aarch64_linux_8u302b08_openj9-0.27.0.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-linux-aarch64-openj9%2F166%2FSemeru-debugimage_aarch64_linux_8u302b08_openj9-0.27.0.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-linux-aarch64-openj9%2F166%2FSemeru-jdk_aarch64_linux_8u302b08_openj9-0.27.0.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-linux-aarch64-openj9%2F166%2FSemeru-jre_aarch64_linux_8u302b08_openj9-0.27.0.tar.gz&amp;BUILD_IDENTIFIER=&amp;NON_AQA_TEST_REPOS_HELP_TEXT=&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=true&amp;NON_AQA_TEST_REPOS=&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;TEST_REPO_PARAMS=&amp;TEST_SELECTION_PARAMS=&amp;TEST_PARALLELIZATION_PARAMS_HELP_TEXT=&amp;JDK_SELECTION_PARAMS=&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=&amp;JDK_VERSION=8&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk8&amp;PLATFORM_AND_MACHINE_HELP_TEXT=&amp;RELEASE_TAG=v0.27.0-release&amp;OPENJ9_BRANCH=v0.27.0-release&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CLOUD_PROVIDER=&amp;PLATFORM_AND_MACHINE=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=v0.27.0-release&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;POST_RUN_PARAMS_HELP_TEXT=&amp;UPSTREAM_TEST_JOB_NUMBER=80&amp;DOCKERIMAGE_TAG=&amp;JDK_SELECTION_PARAMS_HELP_TEXT=&amp;JDK_IMPL=openj9&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=&amp;TKG_SHA=&amp;TEST_SELECTION_PARAMS_HELP_TEXT=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=4e18ffe7-b1b1-4272-9979-99769b68bcc2&amp;ARCHIVE_TEST_RESULTS=false&amp;OPENJDK_SHA=&amp;NUM_MACHINES=&amp;TRSS_URL=http%3A%2F%2Ftrss1.fyre.ibm.com&amp;BUILD_LIST=functional&amp;UPSTREAM_JOB_NUMBER=&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TEST_REPO_PARAMS_HELP_TEXT=&amp;POST_RUN_PARAMS=&amp;TIME_LIMIT=20&amp;TEST_OPTIONS_PARAMS_HELP_TEXT=&amp;JVM_OPTIONS=&amp;PARALLEL=None)

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2021-07-21T17:59:49.508Z] Running test jit_jitt_array_compress_3 ...
[2021-07-21T17:59:49.508Z] ===============================================
[2021-07-21T17:59:49.508Z] jit_jitt_array_compress_3 Start Time: Wed Jul 21 13:59:48 2021 Epoch Time (ms): 1626890388978
[2021-07-21T17:59:49.508Z] "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:destroyAll; "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
[2021-07-21T17:59:49.508Z] JVMSHRC005I No shared class caches available
[2021-07-21T17:59:49.508Z] JVMSHRC005I No shared class caches available
[2021-07-21T17:59:49.508Z] cache cleanup done
[2021-07-21T17:59:49.508Z] variation: -Xjit:disableAsyncCompilation,count=1 -XXgc:forcedShiftingCompressionAmount=4 -Xmx512m
[2021-07-21T17:59:49.509Z] JVM_OPTIONS:  -Xjit:disableAsyncCompilation,count=1 -XXgc:forcedShiftingCompressionAmount=4 -Xmx512m 

[2021-07-21T18:00:03.916Z] Unhandled exception
[2021-07-21T18:00:03.916Z] Type=Segmentation error vmState=0x00000000
[2021-07-21T18:00:03.916Z] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
[2021-07-21T18:00:03.916Z] Handler1=0000FFFF9DD2E6B8 Handler2=0000FFFF9DCA5660 InaccessibleAddress=0000000000000008
[2021-07-21T18:00:03.916Z] R0=0000000000000000 R1=0000000000030F00 R2=0000000000000020 R3=0000000040000000
[2021-07-21T18:00:03.916Z] R4=0000FFFF87927980 R5=0000FFFF980C7370 R6=0000000000000001 R7=0000FFFF9DED0420
[2021-07-21T18:00:03.916Z] R8=0000000000000001 R9=0000000000000002 R10=00000000000CA458 R11=0000FFFF87936E40
[2021-07-21T18:00:03.916Z] R12=000000000002FD00 R13=0000000FE0025F40 R14=00000000000BC690 R15=0000FFFF9D06F624
[2021-07-21T18:00:03.916Z] R16=0000FFFF9CC7BC60 R17=0000FFFF9EAFA800 R18=0000000000000000 R19=0000FFFF4D485890
[2021-07-21T18:00:03.916Z] R20=0000000000040920 R21=0000000FE001F2C0 R22=000000001FE95F28 R23=0000000FFFE95F90
[2021-07-21T18:00:03.916Z] R24=000000000040E000 R25=00000000000CD8F0 R26=0000000000251CF0 R27=0000FFFF9DE9AC08
[2021-07-21T18:00:03.916Z] R28=0000FFFF9DD7D6E0 R29=0000FFFF9EA40F00 R30=0000FFFF9DE100E8 R31=0000FFFF9EA40F00
[2021-07-21T18:00:03.916Z] PC=0000FFFF9DD888A4 SP=0000FFFF9EA40F00 PSTATE=0000000020000000
[2021-07-21T18:00:03.916Z] V0 0000000000000005 (f: 5.000000, d: 2.470328e-323)
[2021-07-21T18:00:03.916Z] V1 0000000000000001 (f: 1.000000, d: 4.940656e-324)
[2021-07-21T18:00:03.916Z] V2 000000003f800000 (f: 1065353216.000000, d: 5.263544e-315)
[2021-07-21T18:00:03.916Z] V3 bfe030d48c3d5a2c (f: 2352830976.000000, d: -5.059607e-01)
[2021-07-21T18:00:03.916Z] V4 bfd00ea348b88334 (f: 1220051712.000000, d: -2.508934e-01)
[2021-07-21T18:00:03.916Z] V5 3fd5575b0be00b6a (f: 199232368.000000, d: 3.334568e-01)
[2021-07-21T18:00:03.916Z] V6 bfdffffef20a4123 (f: 4060758272.000000, d: -4.999997e-01)
[2021-07-21T18:00:03.916Z] V7 3fe62e42fefa39ef (f: 4277811712.000000, d: 6.931472e-01)
[2021-07-21T18:00:03.916Z] V8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V16 bff0000000000000 (f: 0.000000, d: -1.000000e+00)
[2021-07-21T18:00:03.916Z] V17 aaa8000000000000 (f: 0.000000, d: -3.348596e-103)
[2021-07-21T18:00:03.916Z] V18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V19 0000ffff9ea442f0 (f: 2661565184.000000, d: 1.390663e-309)
[2021-07-21T18:00:03.916Z] V20 0000ffff9ea442e0 (f: 2661565184.000000, d: 1.390663e-309)
[2021-07-21T18:00:03.916Z] V21 0000ffff9ea442d0 (f: 2661565184.000000, d: 1.390663e-309)
[2021-07-21T18:00:03.916Z] V22 0000ffff9ea442c0 (f: 2661565184.000000, d: 1.390663e-309)
[2021-07-21T18:00:03.916Z] V23 00000000000000f0 (f: 240.000000, d: 1.185758e-321)
[2021-07-21T18:00:03.916Z] V24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] V31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-07-21T18:00:03.916Z] Module=/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/openjdkbinary/j2sdk-image/jre/lib/aarch64/default/libj9vm29.so
[2021-07-21T18:00:03.916Z] Module_base_address=0000FFFF9DCFA000
[2021-07-21T18:00:03.916Z] Target=2_90_20210721_166 (Linux 4.15.0-141-generic)
[2021-07-21T18:00:03.916Z] CPU=aarch64 (8 logical CPUs) (0x3ea5d4000 RAM)
[2021-07-21T18:00:03.916Z] ----------- Stack Backtrace -----------
[2021-07-21T18:00:03.916Z] (0x0000FFFF9DCA22B8 [libj9prt29.so+0x262b8])
[2021-07-21T18:00:03.916Z] ---------------------------------------
[2021-07-21T18:00:03.916Z] JVMDUMP039I Processing dump event "gpf", detail "" at 2021/07/21 13:59:59 - please wait.
[2021-07-21T18:00:03.916Z] JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/core.20210721.135959.8722.0001.dmp' in response to an event
[2021-07-21T18:00:03.916Z] JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P %E" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.8753.
[2021-07-21T18:00:03.916Z] 
[2021-07-21T18:00:03.916Z] JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/core.20210721.135959.8722.0001.dmp
[2021-07-21T18:00:03.916Z] JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/javacore.20210721.135959.8722.0002.txt' in response to an event
[2021-07-21T18:00:03.916Z] JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/javacore.20210721.135959.8722.0002.txt
[2021-07-21T18:00:03.916Z] JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/Snap.20210721.135959.8722.0003.trc' in response to an event
[2021-07-21T18:00:03.916Z] JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/Snap.20210721.135959.8722.0003.trc
[2021-07-21T18:00:03.916Z] JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/jitdump.20210721.135959.8722.0004.dmp' in response to an event
[2021-07-21T18:00:03.916Z] JVMDUMP051I JIT dump occurred in 'main' thread 0x0000000000011200
[2021-07-21T18:00:03.916Z] JVMDUMP053I JIT dump is recompiling java/lang/invoke/MethodType.createMethodDescriptorString()Ljava/lang/String;
[2021-07-21T18:00:03.916Z] JVMDUMP053I JIT dump is recompiling java/lang/invoke/MethodType.intern()Ljava/lang/invoke/MethodType;
[2021-07-21T18:00:03.916Z] JVMDUMP053I JIT dump is recompiling java/lang/invoke/MethodType.methodType(Ljava/lang/Class;Ljava/util/List;)Ljava/lang/invoke/MethodType;
[2021-07-21T18:00:03.916Z] JVMDUMP053I JIT dump is recompiling java/lang/invoke/MethodType.fromMethodDescriptorString(Ljava/lang/String;Ljava/lang/ClassLoader;)Ljava/lang/invoke/MethodType;
[2021-07-21T18:00:03.916Z] JVMDUMP053I JIT dump is recompiling java/lang/invoke/MethodTypeHelper.vmResolveFromMethodDescriptorString(Ljava/lang/String;Ljava/lang/ClassLoader;Ljava/lang/Class;)Ljava/lang/invoke/MethodType;
[2021-07-21T18:00:03.916Z] JVMDUMP053I JIT dump is recompiling java/lang/invoke/MethodHandleResolver.sendResolveMethodHandle(ILjava/lang/Class;Ljava/lang/Class;Ljava/lang/String;Ljava/lang/String;Ljava/lang/ClassLoader;)Ljava/lang/invoke/MethodHandle;
[2021-07-21T18:00:03.916Z] JVMDUMP053I JIT dump is recompiling java/lang/invoke/MethodHandleResolver.getAdditionalBsmArg(Lcom/ibm/oti/vm/VMLangAccess;Ljava/lang/Object;Ljava/lang/Class;Ljava/lang/invoke/MethodHandle;JII)Ljava/lang/Object;
[2021-07-21T18:00:08.197Z] JVMDUMP010I JIT dump written to /home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_aarch64_linux_testList_1/aqa-tests/TKG/output_16268897074933/jit_jitt_array_compress_3/jitdump.20210721.135959.8722.0004.dmp
[2021-07-21T18:00:08.197Z] JVMDUMP013I Processed dump event "gpf", detail "".
[2021-07-21T18:00:08.197Z] 
[2021-07-21T18:00:08.197Z] jit_jitt_array_compress_3_FAILED
```

Launched a 10x grinder (3 machines) at `Grinder/17075`

fyi @knn-k

**Comment by JasonFengJ9** [comment]
Also seen at JDK11 `Test_openjdk11_j9_sanity.functional_aarch64_linux_testList_0/23/` (`cent7-aarch64-4`).
```
openjdk version "11.0.12" 2021-07-20
OpenJDK Runtime Environment Temurin-11.0.12+7 (build 11.0.12+7)
Eclipse OpenJ9 VM Temurin-11.0.12+7 (build openj9-0.27.0, JRE 11 Linux aarch64-64-Bit Compressed References 20210721_185 (JIT enabled, AOT enabled)
OpenJ9   - 1851b0074
OMR      - 9db1c870d
JCL      - 134c165477 based on jdk-11.0.12+7)
```

**Comment by knn-k** [comment]
This looks similar to #12940 and #12945.

**Comment by knn-k** [comment]
The fix #12946 is not applied to the 0.27 branch.

**Comment by JasonFengJ9** [comment]
> This looks similar to #12940 and #12945.
> The fix #12946 is not applied to the 0.27 branch.

Agreed, do we want 0.27 release to have this fix? what's the level of risk?

fyi @keithc-ca

**Comment by knn-k** [comment]
I opened #13231 for fixing this.

**Comment by knn-k** [comment]
Please close this issue, although the fix was not merged to the 0.27 branch.

**Comment by JasonFengJ9** [comment]
0.27 is finalized, closing..
